source/adapters/native_cpu/kernel.hpp

```diff
@@ -64,27 +55,62 @@ struct ur_kernel_handle_t_ : RefCounted {
     std::vector<bool> OwnsMem;
     static constexpr size_t MaxAlign = 16 * sizeof(double);
 
+    arguments() = default;
+
+    arguments(const arguments &Other)
+        : Indices(Other.Indices), ParamSizes(Other.ParamSizes),
+          OwnsMem(Other.OwnsMem.size()) {
```

Comment:
In C++03 this would have guaranteed `OwnsMem` is a vector of `false`, but in C++11 it does not, this probably needs to be `OwnsMem(Other.OwnsMem.size(), false)` instead.